functions/completed-checks/v1.js

```diff
@@ -0,0 +1,259 @@
+'use strict'
+
+const moment = require('moment')
+const sqlService = require('less-tedious')
+const uuid = require('uuid/v4')
+const { TYPES } = require('tedious')
+
+const config = require('../config')
+sqlService.initialise(config)
+const azureStorageHelper = require('../lib/azure-storage-helper')
+const sqlUtil = require('../lib/sql-helper')
+const markingService = require('./marking.service')
+
+// SQL server vars
+const checkResultTable = '[checkResult]'
+const schema = '[mtc_admin]'
+const checkStatusTable = '[checkStatus]'
+const checkTable = '[check]'
+
+const v1 = {
+  process: async function (context, completedCheckMessage) {
+    await handleCompletedCheck(context, completedCheckMessage)
+  }
+}
+
+// Table Storage
+const azureTableService = azureStorageHelper.getPromisifiedAzureTableService()
+
+async function handleCompletedCheck (context, completedCheckMessage) {
+  context.log('completed-check: message received', completedCheckMessage.checkCode)
+
+  let checkData
+  try {
+    checkData = await sqlUtil.sqlFindCheckWithFormDataByCheckCode(completedCheckMessage.checkCode)
+  } catch (error) {
+    context.log.error(`completed-check: ERROR: failed to retrieve checkData for checkCode: ${completedCheckMessage.checkCode}`)
+    throw error
+  }
+
+  if (!canCompleteCheck(checkData.code)) {
+    const errorMessage = `completed-check: ERROR: check ${completedCheckMessage.checkCode} is not in a correct state to be completed. Current state is ${checkData.code}`
```

Comment:
seems a bit odd to be using `code` to describe check state, i would assume that was the checkCode.  something more obvious such as `checkData.currentState` would read better.

Reply:
@GuyHarwood since it is essentially a code I named it `checkStatusCode`.